sys/kern/sysv_sem.c

```diff
@@ -2210,7 +2125,8 @@ freebsd7_freebsd64___semctl(struct thread *td,
 		break;
 	case GETALL:
 	case SETALL:
-		semun.array = __USER_CAP_ARRAY(arg.array, SHRT_MAX);
+		semun.array = __USER_CAP(arg.array, seminfo.semmns *
```

Comment:
Technically this is an int so can be larger than SHRT_MAX so long as you don't expect to use all of them in a single process, but in practice it's way smaller than SHRT_MAX.